service/controller/resource/cleanupvpcpeerings/delete.go

```diff
@@ -0,0 +1,123 @@
+package cleanupvpcpeerings
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/controller/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/key"
+)
+
+func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	cl, err := key.ToCluster(ctx, obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Fetch all VPC peering connections of the Tenant Cluster VPC
+	var vpcPeeringConnectionIDs []*string
+	{
+		r.logger.Debugf(ctx, "finding all vpc peering connections")
+
+		// fetch all vpc peering connection via requester vpc filter
+		{
+			i := &ec2.DescribeVpcPeeringConnectionsInput{
+				Filters: []*ec2.Filter{
+					{
+						Name: aws.String("requester-vpc-info.vpc-id"),
+						Values: []*string{
+							aws.String(cl.Status.Provider.Network.VPCID),
+						},
+					},
+				},
+			}
+
+			o, err := cc.Client.TenantCluster.AWS.EC2.DescribeVpcPeeringConnections(i)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			for _, s := range o.VpcPeeringConnections {
+				if isTCCPVPCPeering(s.Tags) {
+					// skip deleting vpc peering connection from tccp CF stack
+					// it will be deleted by the CF
+					continue
+				}
+				if *s.Status.Code == vpcStatusDeleting || *s.Status.Code == vpcStatusDeleted {
+					// ignore vpc peering connections that have status deleting or deleted
+					continue
+				}
+
+				vpcPeeringConnectionIDs = append(vpcPeeringConnectionIDs, s.VpcPeeringConnectionId)
+			}
+		}
+		// fetch all vpc peering connection via accepter vpc filter
+		{
+			i := &ec2.DescribeVpcPeeringConnectionsInput{
+				Filters: []*ec2.Filter{
+					{
+						Name: aws.String("accepter-vpc-info.vpc-id"),
```

Comment:
does it make sense do have it in a small helper function? it seems it's duplicate code except the name is different.

Reply:
I have adjusted the code, can you check now?